### PR TITLE
Hotfix: Return empty dict when no highlight dict is present

### PR DESCRIPTION
**Pull request by dojutsu-user**
I have moved the sorting logic to `search.utils`.

### CODE REVIEW

**Comment by dojutsu-user** [comment]
Ohhh... I just noticed.
We have modified the parse_json to remove all the new-line characters.
If we don't have any new line characters in our index -- then there won't be any new line characters in the highlight.
I think the func `_remove_newlines_from_dict()` is no longer required.

**Comment by dojutsu-user** [comment]
This should also increase the speed as that func is going through every highlight dict recursively and that too -- doing nothing.

File: readthedocs/search/utils.py [hunk 1]
```diff
@@ -153,24 +155,27 @@ def _indexing_helper(html_objs_qs, wipe=False):
...
-        else:
-            # elastic returns the contents of the
-            # highlighted field in a list.
-            if isinstance(v, list):
-                v_new_list = [res.replace('\n', '. ') for res in v]
-                highlight[k] = v_new_list
-
-    return highlight
+        if logging:
+            log.debug('API Search highlight: %s', pformat(highlight_dict))
```
Comment (dojutsu-user) [inline readthedocs/search/utils.py]: I also wanted to know that if we can do something for `pformat`.
@humitos pointed out that it is doing unnecessary work ([comment](https://github.com/readthedocs/readthedocs.org/pull/5919#discussion_r303579977)).
There are other places also where we are using `pformat` for logging purposes.
This must be reducing some speed and taking memory.

Comment (ericholscher) [reply readthedocs/search/utils.py]: We can just output it unformatted.

File: readthedocs/search/api.py [hunk 2]
```diff
@@ -48,27 +47,13 @@ def get_inner_hits(self, obj):
...
-                    '_source': hit._source.to_dict(),
-                    'highlight': self._get_inner_hits_highlights(hit),
-                }
-                for hit in sorted(all_results, key=attrgetter('_score'), reverse=True)
-            ]
-
+            sorted_results = list(utils._get_sorted_results(
+                results=all_results,
+                source_key='_source',
+                logging=True,
```
Comment (ericholscher) [inline readthedocs/search/api.py]: optional logging feels weird. Do we ever not want logs?

Comment (ericholscher) [reply readthedocs/search/api.py]: I think we should probably just log outside this function, and remove the logging from this function call.

File: readthedocs/search/utils.py [hunk 3]
```diff
@@ -153,24 +155,27 @@ def _indexing_helper(html_objs_qs, wipe=False):
...
+
+        return highlight_dict
+    return {}
+
+
+def _get_sorted_results(results, source_key='_source', logging=False):
+    """Sort results according to their score and return a generator expression."""
+    sorted_results = (
```
Comment (ericholscher) [inline readthedocs/search/utils.py]: Do we really need this to be a generator? What is the goal of it versus just using a list?  It looks like we're just calling `list` on it both times we use it, which means it should probably just return a list :)

File: readthedocs/search/utils.py [hunk 4]
```diff
@@ -153,24 +155,27 @@ def _indexing_helper(html_objs_qs, wipe=False):
                 delete_objects_in_es.delay(**kwargs)
 
 
-def _remove_newlines_from_dict(highlight):
-    """
-    Recursively change results to turn newlines into periods.
+def _get_inner_hits_highlights(hit, logging=False):
```
Comment (ericholscher) [inline readthedocs/search/utils.py]: It seems like maybe we don't even need this function. It's mostly just checking if the hit has a highlight and returning it.

Comment (dojutsu-user) [reply readthedocs/search/utils.py]: Remvoed. :+1:

File: readthedocs/search/api.py (hunk 2 above)
Comment (dojutsu-user) [reply readthedocs/search/api.py]: Done.

File: readthedocs/search/utils.py (hunk 3 above)
Comment (dojutsu-user) [reply readthedocs/search/utils.py]: I thought that returning a generator expression would be efficient than returning a huge list of results?

**Comment by ericholscher** [comment]
Looks good once we fix the conflict 👍

**Pull request merged**